src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -374,14 +379,15 @@ public void fireFileStarted(String fileName) {
      * Notify all listeners about the errors in a file.
      *
      * @param fileName the audited file
-     * @param errors the audit errors from the file
+     * @param fileProcessingResult file processing results object
      */
     @Override
-    public void fireErrors(String fileName, SortedSet<LocalizedMessage> errors) {
+    public void fireErrors(String fileName, FileProcessingResult fileProcessingResult) {
         final String stripped = CommonUtils.relativizeAndNormalizePath(basedir, fileName);
         boolean hasNonFilteredViolations = false;
-        for (final LocalizedMessage element : errors) {
-            final AuditEvent event = new AuditEvent(this, stripped, element);
+        final SortedSet<LocalizedMessage> errorMessages = fileProcessingResult.getMessages();
+        for (LocalizedMessage errorMessage : errorMessages) {
+            final AuditEvent event = new AuditEvent(this, stripped, errorMessage);
```

Comment:
@rnveach 
You are right. AuditEvent should receive FileProcessingResult. Changes the plan.
https://github.com/checkstyle/checkstyle/issues/3573#issuecomment-308438894